packages/ListBox/src/components/Trigger/Trigger.js

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import PropTypes from "prop-types";
+import RawButton from "@paprika/raw-button";
+import Label from "../Label";
+import handleKeyboardKeys from "../../helpers/handleKeyboardKeys";
+import useListBox from "../../useListBox";
+import applyOnChange, {
+  cleanActionTypes,
+  getSelectedOptionSingle,
+  getSelectedOptionsMulti,
+} from "../../helpers/applyOnChange";
+
+import {
+  ListBoxTriggerStyled,
+  TriggerArrowStyled,
+  RawButtonClearButtonStyled,
+  TriggerActionIconsContainer,
+} from "./Trigger.styles";
+import { getDOMAttributesForListBoxButton } from "../../helpers/DOMAttributes";
+
+const propTypes = {
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+  hasClearButton: PropTypes.bool,
+  onClickClear: PropTypes.func,
+  onFooterClickAccept: PropTypes.func,
+  placeholder: PropTypes.string,
+};
+
+const defaultProps = {
+  children: <React.Fragment />,
+  hasClearButton: true,
+  onClickClear: null,
+  onFooterClickAccept: null,
+  placeholder: "Select...",
+};
+
+export default function Trigger(props) {
+  const [state, dispatch] = useListBox();
+  const { placeholder, hasClearButton, onFooterClickAccept } = props;
+
+  const { isDisabled, refTriggerContainer, refTrigger, isMulti } = state;
+
+  const handleClick = () => {
+    if (isDisabled) {
+      return;
+    }
+
+    dispatch({ type: useListBox.types.togglePopover });
+  };
+
+  const handleClickClear = () => {
+    if (isDisabled) {
+      return;
+    }
+
+    if (props.onClickClear) {
+      props.onClickClear(state, dispatch);
+      return;
+    }
+
+    if (state.hasFooter) {
+      dispatch({
+        type: useListBox.types.clear,
+        payload: {
+          isOpen: false,
+          onChangeFn: applyOnChange(onFooterClickAccept, "listbox:footer:clear"),
+        },
+      });
+      return;
```

Comment:
Why do you hate `else` so much? 😄

Reply:
I don't find it useful in most cases if you have the power of return :P. Totally a preference thing.